MMTabBarView/MMTabBarView/MMTabBarView.m

```diff
@@ -1318,6 +1318,13 @@ - (void)hideTabBar:(BOOL)hide animate:(BOOL)animate {
 			}
 		}
 	}
+	[self applyFrameChangesAnimated:animate hide:hide
+		partnerTargetOrigin:partnerTargetOrigin partnerTargetSize:partnerTargetSize
+		completion:^{
+			[self setHidden:hide];
```

Comment:
This line was split in the original code; the hidden variable was set to true immediately but to false only after the animation. Usually, delaying an update of the underlying state to wait for asynchronous gui changes is no good idea, but what do you think about it?

Reply:
Looking at the code, I cannot find any justification for an asymmetry between hide and show: both actions may involve an animation, and both may handle delegate calls.

I don't think there's any particular risk, as all these NSView resizing have to be called from the main thread and our own code does not dispatch any auxiliary thread.

Reply:
My guess is that this is about things happening in quick succession, such as closing a tab and pressing ctrl+t while the animation is still running. In that case we would probably create a tab but not change the tabbar size, because hidden is still false.

Reply:
However, such user actions would be queued on the main thread, so the subsequent actions would be executed in the expected order.

See also that all our internal tests are based on our own internal variables  (for instance, `_isHidden` and `_shouldDisplayTabBar`) instead of NSView's status, and these internal variables are set ahead of any execution :  https://github.com/barijaona/MMTabBarView/blob/9124d195381b403d15d9c72721f5acf2b21f6457/MMTabBarView/MMTabBarView/MMTabBarView.m#L1205-L1211

To be sure, I hacked the demo app in order to simulate the succession of user's action you described : everything went as expected.

Reply:
Seems like I had a misconception about the asynchronous execution of animations. Thanks for testing and sorry for the hassle :)